osu.Server.Queues.ScoreStatisticsProcessor/Stores/BeatmapStore.cs

```diff
@@ -29,14 +29,27 @@ public class BeatmapStore
     {
         private static readonly bool use_realtime_difficulty_calculation = Environment.GetEnvironmentVariable("REALTIME_DIFFICULTY") != "0";
         private static readonly string beatmap_download_path = Environment.GetEnvironmentVariable("BEATMAP_DOWNLOAD_PATH") ?? "https://osu.ppy.sh/osu/{0}";
+        private static readonly uint memory_cache_size_limit = uint.Parse(Environment.GetEnvironmentVariable("MEMORY_CACHE_SIZE_LIMIT") ?? "1000");
```

Comment:
this unit is super counter intuitive. if it can't be in megabytes then i'm against this change.

Reply:
yeah this can be MB or really any unit that you want, since we can just do some math with the size of the classes.

Reply:
> but maybe we can do a `sizeof(T)` summation for each field in the class somewhere

At this point I'd probably rather hardcode a number with an inline comment or something.

Do mind that in the case of difficulty attributes the size will depend on the ruleset, so that'll have to be like looked up from a dictionary or something.

Reply:
> Do mind that in the case of difficulty attributes the size will depend on the ruleset

I don't think this applies here, regardless of ruleset the same [`BeatmapDifficultyAttribute`](https://github.com/ppy/osu-queue-score-statistics/blob/06bbe2015dc16c4a556b0ee3b944b75dbfb5a5a9/osu.Server.Queues.ScoreStatisticsProcessor/Models/BeatmapDifficultyAttribute.cs#L13-L28) is fetched

Reply:
Oh I confused that with the client side classes. Sure.

Reply:
What I was going to post initially was something along these lines:

As bytes you optimise for size, but this is an estimation at best that doesn't account for fragmentation, heap sizes, and possibly other factors. It's also only realistically a lower bound because it doesn't factor burst usage (e.g. while a diffcalc goes on, perhaps). There's no real good way to measure this afaik - I did it visually based on what I know about C#.

On the other hand "1" allows you to think of it as the number of most recently used objects. You'd tune this directly based on hit/miss ratios to reach a target, and tune k8s to give the container enough resources after that.
In a way, the counterintuitiveness of it makes more intuitive.

Reply:
Hmm, that might make sense from one end, but when deploying, we'd have to be doing manual calculations to figure out memory requirements for the deploy. Generally we don't decide cache limits based on hit ratio but by available resources on the server.

Feels weird having a `MemoryCache` component which doesn't guarantee an upper limit though..